torch/distributed/fsdp/_optim_utils.py

```diff
@@ -1280,8 +1275,20 @@ def _optim_state_dict(
     The flattened parameters in ``FSDP`` modules contained in ``model``
     are mapped back to their unflattened parameters.
 
+    Parameter keys are not well-defined. For a regular optimizer, the optimizer
+    state_dict contains a mapping from parameter IDs to parameter states.
+    Parameter IDs are the order of parameters in ``optim.param_groups()`` across
+    all the groups. This API also allows user to pass ``optim_input`` for the
+    mapping between parameters and parameter IDs. Using ``optim_input`` is being
+    deprecated.
+
+    If the optimizer is a ``NamedOptimizer``, the optimizer state_dict does not
```

Comment:
what if optim_input is provided but also it is a NamedOptimizer, will that create an issue?

Reply:
Yes, it will fail. Will add a error handling for this.